### PR TITLE
Fix QR not mapping answers

### PR DESCRIPTION
**Pull request by FikriMilano**
**IMPORTANT: Where possible all PRs must be linked to a Github issue**

Fixes #[issue number] or Closes #[issue number]

Changes:
- Fix QR not found in DB
- Remove initialExpression from a Questionnaire Item if that Questionnaire Item already has configurable pre-population values
- Fix FHIR Path initialExpressions for Family Questionnaire and Family Member Questionnaire
- Refactor ProfileViewModel and QuestionnaireActivity to error handling in questionnaire flow

**Engineer Checklist**
- [ ] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [ ] I have added any strings visible on UI components to the `strings.xml` file
- [ ] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [ ] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [ ] I have built and run the FHIRCore app to verify my change fixes the issue and/or does not break the app 

**Code Reviewer Checklist**
- [ ] I have verified **Unit tests** have been written for any new feature(s) and edge cases
- [ ] I have verified any strings visible on UI components are in the `strings.xml` file
- [ ] I have verifed the [CHANGELOG.md](./CHANGELOG.md) file has any notable changes to the codebase
- [ ] I have verified the solution has been implemented in a configurable and generic way for reuseable components
- [ ] I have built and run the FHIRCore app to verify the change fixes the issue and/or does not break the app

### CODE REVIEW

**Comment by FikriMilano** [comment]
@dubdabasoduba can you guide me which questionnaire I should update in the fhir-resource repo?
I can see 2 different version that are using these questionnaires:
1. ecbis_liberia https://github.com/opensrp/fhir-resources/blob/main/ecbis_liberia/questionnaire/patient_registration/family_member_registration.json
2. mcgovern https://github.com/opensrp/fhir-resources/blob/main/mcgovern/questionnaire/patient_registration/family_member_registration.json

**Comment by f-odhiambo** [comment]
> 1. ecbis_liberia https://github.com/opensrp/fhir-resources/blob/main/ecbis_liberia/questionnaire/patient_registration/family_member_registration.json

@FikriMilano You can use this one - ecbis_liberia https://github.com/opensrp/fhir-resources/blob/main/ecbis_liberia/questionnaire/patient_registration/family_member_registration.json  but just do a PR and tag for review

**Review by pld — COMMENTED** [review]
looks good! There are still a number of code paths in ProfileViewModel to add tests for

File: android/engine/src/main/java/org/smartregister/fhircore/engine/util/extension/QuestionnaireExtension.kt [hunk 1]
```diff
@@ -136,13 +136,18 @@ fun List<Questionnaire.QuestionnaireItemComponent>.find(
...
+  "http://hl7.org/fhir/uv/sdc/StructureDefinition/sdc-questionnaire-initialExpression"
+
 /** Pre-Populate Questionnaire items with initial values */
 // TODO: handle interpolation for null values on rules engine and not where the values are used
 fun List<Questionnaire.QuestionnaireItemComponent>.prePopulateInitialValues(
   interpolationPrefix: String,
   prePopulationParams: List<ActionParameter>
 ) {
   forEach { item ->
+    if (item.hasExtension(ITEM_INITIAL_EXPRESSION_URL))
```
[the remark this answers is not in the document]
Comment (FikriMilano) [reply android/engine/src/main/java/org/smartregister/fhircore/engine/util/extension/QuestionnaireExtension.kt:149]: just one sec

**Comment by FikriMilano** [comment]
@pld yeah, I only add the top edge cases for now. Then add the rest during CHT, is that ok?
cc @f-odhiambo

**Comment by FikriMilano** [comment]
@pld thanks, beware though, the mocks are frustrating :D haha

**Comment by pld** [comment]
Haha oh yea I am well aware :D

File: android/quest/src/test/java/org/smartregister/fhircore/quest/ui/profile/ProfileViewModelTest.kt [hunk 2]
```diff
@@ -183,38 +211,123 @@ class ProfileViewModelTest : RobolectricTest() {
...
+        resourceData = null,
+        overflowMenuItemConfig = overflowMenuItemConfig,
+      )
+
+    runBlocking {
+      profileViewModel.onEvent(event)
+      delay(100)
+    }
+
+    Assert.assertNotNull(profileViewModel.snackBarStateFlow)
```
Comment (pld) [inline android/quest/src/test/java/org/smartregister/fhircore/quest/ui/profile/ProfileViewModelTest.kt:324]: I added this, but it would be better to test the actual error in the snackBarState, work for CHT

**Review by pld — APPROVED** [review]

File: android/quest/src/test/java/org/smartregister/fhircore/quest/ui/profile/ProfileViewModelTest.kt (hunk 2 above)
Comment (FikriMilano) [reply android/quest/src/test/java/org/smartregister/fhircore/quest/ui/profile/ProfileViewModelTest.kt:324]: yep, noted
